components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/dao/RequestObjectDAOImpl.java

```diff
@@ -279,6 +279,49 @@ private void insertRequestObjectClaimValues(int requestObjectClaimId, List<Strin
         return essentialClaims;
     }
 
+    /**
+     * Retrieve essential claims for the id token and user info endpoint.
+     *
+     * @param sessionDataKey session Data Key
+     * @param isUserInfo     return true if the claims are requested from user info end point.
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    @Override
+    public List<RequestedClaim> getRequestedClaimsBySessionDataKey(String sessionDataKey, boolean isUserInfo)
+            throws IdentityOAuth2Exception {
+        Connection connection = null;
+        PreparedStatement prepStmt = null;
+        ResultSet resultSet = null;
+        List<RequestedClaim> essentialClaims = new ArrayList<>();
+        try {
+            connection = IdentityDatabaseUtil.getDBConnection();
```

Comment:
Shall we use try-with-resources statements like in [1].
It will ensure that each resource is closed at the end of the statement without us explicitly doing that.


[1]https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/5.6.x/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java#L1805

Reply:
Thanks @denuwanthi . I was just been consistent with the method structure followed throughout the class. Will make use of your suggestion.